pkgs/applications/networking/instant-messengers/zulip-desktop/default.nix

```diff
@@ -0,0 +1,71 @@
+{ appimageTools, stdenv, fetchurl, unzip } :
+
+let
+  inherit (stdenv.hostPlatform) system;
+
+  plat = {
+    "x86_64-linux" = "-x86_64";
+    "x86_64-darwin" = "";
+  }.${system};
+
+  sha256 = {
+    "x86_64-linux" = "1pni02mb5bvwx3k45vd6ga269ghdl633gjklyslai24rrhp16h9z";
+    "x86_64-darwin" = "1x2gxixir766670b5iglcgzbzmjap1k0n0nhpjhkqvn2ngs8fzq4";
+  }.${system};
+
+  archive_fmt = if system == "x86_64-darwin" then "zip" else "AppImage";
+
+  pname = "zulip-desktop";
+  imageName = "Zulip";
+  version = "4.0.0";
+  executablename = "${imageName}-${version}${plat}";
+
+  src = fetchurl {
+    url = "https://github.com/zulip/${pname}/releases/download/v${version}/${executablename}.${archive_fmt}";
```

Comment:
Have you tried building it from source? Seems like an electron app to me. See:

- https://github.com/zulip/zulip-desktop/blob/v4.0.0/development.md
- https://github.com/zulip/zulip-desktop/blob/v4.0.0/package.json#L23